buildSrc/src/main/groovy/io/deephaven/tools/License.groovy

```diff
@@ -111,4 +91,52 @@ class License {
         Jar jar = project.tasks.findByName('jar') as Jar
         jar.from(processLicenseResourcesTask.outputs)
     }
+
+    TaskProvider<Sync> syncSourceSetLicense(String licenseSourceSetDir) {
+        def copyrightYear = '2021'
+        syncLicensesProvider(
+                project,
+                'syncSourceSetLicense',
+                "${licenseSourceSetDir}/META-INF",
+                license.name,
+                notice.name,
+                copyrightYear,
+                project.name)
+    }
+
+    TaskProvider<Sync> syncDockerLicense() {
+        def copyrightYear = '2021'
+        syncLicensesProvider(
+                project,
+                'syncDockerLicense',
+                "${project.buildDir}/syncDockerLicense",
+                license.name,
+                notice.name,
+                copyrightYear,
+                project.name)
+    }
+
+    private TaskProvider<Sync> syncLicensesProvider(Project project, String taskName, String destDir, String licenseFilename, String noticeFilename, String copyrightYear, String projectName) {
+        project.tasks.register(taskName, Sync) {
+            it.from(license)
+            it.from(notice)
+
+            it.into(destDir)
+
+            it.rename(licenseFilename, OUTPUT_LICENSE_NAME)
+            it.rename(noticeFilename, OUTPUT_NOTICE_NAME)
+
+            it.expand(copyrightYear: copyrightYear, projectName: project.name)
+
+            // Make sure we invalidate this task if the structure changes
+            it.inputs.property('OUTPUT_LICENSE_NAME', OUTPUT_LICENSE_NAME)
```

Comment:
so this is defined earlier in this same class, which means if the constant changes, the classpath will change, and gradle will notice this. You can confirm this by building with `--info`, and only changing that one constant, you should see something like this output
```
> Task :web-client-ide:buildDocker
Caching disabled for task ':web-client-ide:buildDocker' because:
  Build cache is disabled
  Caching has not been enabled for the task
Task ':web-client-ide:buildDocker' is not up-to-date because:
  Class path of task ':web-client-ide:buildDocker' has changed from da03359ad4cbdd6877ca2bae740ef39b to e52252d860e65fecef1085604f14aca7.
```

Of course if those aren't actually constants, but read from configuration or something, that won't work, so it could be better to be defensive... but this is in the same class as the implementation. So this isn't really a nit, but a "this isn't necessary, now you know", change it if you want to.

Reply:
I think I found the opposite - when I was writing License, if I changed the output license name, the task was incorrectly cached. Let me give that another go...

Reply:
So commenting out this input property and running `./gradlew grpc-api:jar` (it should work w/ any jar) to get it built, and then changing OUTPUT_LICENSE_NAME to `"LICENSE.md"`, I get that the task is up-to-date:
```
> Task :grpc-api:compileJava UP-TO-DATE
> Task :grpc-api:processResources NO-SOURCE
> Task :grpc-api:classes UP-TO-DATE
> Task :grpc-api:syncSourceSetLicense UP-TO-DATE
> Task :grpc-api:processLicenseResources UP-TO-DATE
> Task :grpc-api:jar UP-TO-DATE
```

Reply:
And the actual sync task is up to date, even w/ the docker side of things:

`./gradlew grpc-proxy:syncDockerLicense`
```
> Task :grpc-proxy:syncDockerLicense UP-TO-DATE
```

So, even though changing the OUTPUT_LICENSE_NAME will invalidate *some* tasks (b/c they depend on buildSrc on the classpath?), it won't actually sync the correct license names out.

Reply:
That seems terrifying - does any change of any buildSrc require a full clean every time then? What if you change the implementation of a method that is doing work - you can't create an input property for each possible change like that...